components/collector/src/quality_time_collector.py

```diff
@@ -2,18 +2,22 @@
 
 import asyncio
 import logging
+import os
+from typing import NoReturn
 
 # Make sure subclasses are registered
 import metric_collectors  # pylint: disable=unused-import # lgtm [py/unused-import]
 import source_collectors  # pylint: disable=unused-import # lgtm [py/unused-import]
 from base_collectors import Collector
 
 
-async def collect(log_level: int | None = None) -> None:
+async def collect(log_level: int | str | None = None) -> NoReturn:
     """Collect the measurements indefinitely."""
-    logging.getLogger().setLevel(logging.ERROR if log_level is None else log_level)
+    if log_level is None:
+        log_level = str(os.getenv("COLLECTOR_LOG_LEVEL", "WARNING"))
+    logging.getLogger().setLevel(log_level)
```

Comment:
Would `log_level or os.getenv("COLLECTOR_LOG_LEVEL", "WARNING")` not suffice?

Reply:
That would make it impossible to pass 0 as log level, which is an allowed value, see https://docs.python.org/3/library/logging.html#logging-levels.